docs/datatypes/chain.md

```diff
@@ -67,9 +70,9 @@ NonEmptyChain.fromChainPrepend(1, Chain(2, 3))
 ```
 ## How it works
 
-`Chain` is a fairly simple data structure compared to something like `Vector`.
-It's a simple ADT that has only 4 cases.
-It is either an empty `Chain` with no elements, a singleton `Chain` with exactly one element, a concatenation of two chains or a wrapper for another collection.
+`Chain` is implemented as a simple unbalanced binary tree ADT with four cases:
+an empty `Chain` with no elements, a singleton `Chain` with exactly one element, a concatenation of two chains, or a wrapper for another collection.
```

Comment:
I would prefer `or a wrapper for another Seq.` We can't wrap say `Set[Int]` which is a collection.

Reply:
Chain.fromSeq is described a little further down and shows the 0 & 1 cases. Updated to mention Seq.